src/typeahead/typeahead.spec.ts

```diff
@@ -51,360 +51,291 @@ function expectWindowResults(element, expectedResults: string[]) {
   expectResults(window, expectedResults);
 }
 
+function createTestComponentFixture(html: string): ComponentFixture<TestComponent> {
+  TestBed.overrideComponent(TestComponent, {set: {template: html}});
+  const fixture = TestBed.createComponent(TestComponent);
+  fixture.detectChanges();
+  return fixture;
+}
+
 describe('ngb-typeahead', () => {
 
   beforeEach(() => {
-    TestBed.configureTestingModule({imports: [NgbTypeaheadModule, FormsModule, ReactiveFormsModule]});
+    TestBed.configureTestingModule(
+        {declarations: [TestComponent], imports: [NgbTypeaheadModule, FormsModule, ReactiveFormsModule]});
   });
 
   describe('valueaccessor', () => {
 
     // TODO: remove 'whenStable' once 'core/testing' is fixed
-    it('should format values when no formatter provided', async(inject([TestComponentBuilder], (tcb) => {
-         const html = '<input [(ngModel)]="model" [ngbTypeahead]="findNothing" />';
+    it('should format values when no formatter provided', async(() => {
+         const fixture = createTestComponentFixture('<input [(ngModel)]="model" [ngbTypeahead]="findNothing" />');
 
-         tcb.overrideTemplate(TestComponent, html).createAsync(TestComponent).then((fixture) => {
-           fixture.detectChanges();
-           const el = fixture.nativeElement;
-           const comp = fixture.componentInstance;
-           expectInputValue(el, '');
+         const el = fixture.nativeElement;
+         const comp = fixture.componentInstance;
+         expectInputValue(el, '');
 
-           comp.model = 'text';
-           fixture.detectChanges();
-           fixture.whenStable().then(() => {
-             expectInputValue(el, 'text');
+         comp.model = 'text';
+         fixture.detectChanges();
+         fixture.whenStable()
+             .then(() => {
+               expectInputValue(el, 'text');
 
-             comp.model = null;
-             fixture.detectChanges();
-             fixture.whenStable().then(() => {
+               comp.model = null;
+               fixture.detectChanges();
+               return fixture.whenStable();
```

Comment:
I had a hard time understanding promise chaining with angular 1. Now is the time to capitalize. I couldn't resist flattening the pyramid of callbacks :-)